server/src/projects/projects.service.ts

```diff
@@ -39,9 +39,12 @@ export class ProjectsService {
       `);
       return this.overwriteCodesWithLabels(records);
     } catch(e) {
-      const errorMessage = `Error finding projects by contact ID. ${e.message}`;
+      const errorMessage = `Error finding projects by contact ID. CRM: ${e.message}`;
       console.log(errorMessage);
-      throw new HttpException(errorMessage, HttpStatus.UNAUTHORIZED);
+      throw new HttpException({
+        "code": "FIND_MANY_BY_CONTACTID_PROJECT_SERVICE",
+        "message": errorMessage,
```

Comment:
Yeah, I would say this code shouldn't be specific to the named object itself, but more of a generic description.

Something like `PROJECT_NOT_FOUND`. I would make the code try to describe as much as is reasonably inferable from the promise failure. In some other cases you do an `instance of` which is good. Some even more robust error handling might check the CRM error response body: https://docs.microsoft.com/en-us/dynamics-nav/api-reference/v1.0/dynamics_error_codes

Ditto on `message` — it's not an allowed attribute in JSON:API (unless you want to use `meta`, but `message` seems like what `detail` is trying to achieve.